charts/tidb-cluster/values.yaml

```diff
@@ -428,18 +428,25 @@ tidb:
     list: ["whitelist-1"]
 
   # Whether enable TLS connection between TiDB server and MySQL client.
-  # When enabled, TiDB will accept TLS encrypted connections from MySQL client, certificates will be generated
-  # automatically.
   # Note: TLS connection is not forced on the server side, plain connections are also accepted after enableing.
-  enableTLSClient: false
-  # # extra SAN IP list when you set tidb.enableTLSClient to true
-  # extraSANIPList:
-  #   - 1.1.1.1
-  #   - 2.2.2.2
-  # # extra SAN Domain List when you set tidb.enableTLSClient to true
-  # extraSANDomainList:
-  #   - example1.com
-  #   - example2.com
+  tlsClient:
+    # When enabled, TiDB will accept TLS encrypted connections from MySQL client
+    enabled: false
+    # Auto-generated certificate
+    autoGenerated: {}
+      # # extra SAN IP list
+      # extraSANIPList:
+      #  - 1.1.1.1
+      #  - 2.2.2.2
+      # # extra SAN Domain list
+      # extraSANDomainList:
+      #  - example1.com
+      #  - example2.com
+    # User-generated certificate
+    userGenerated: {}
+      # # secretName is the name of the secret which stores user-defined tidb server certificate, key and ca, create it with:
+      # # kubectl create secret generic <secret-name> --namespace=<namespace> --from-file=cert=<tidb server certificate file path> --from-file=key=<tidb server key file path> --from-file=ca=<ca file path>
+      # secretName: "demo-tidb-server-secret"
```

Comment:
I think we can simplify this a bit, like [ingress tls config](https://kubernetes.io/docs/concepts/services-networking/ingress/):

```
  tls:
  - hosts:
    - sslexample.foo.com
    secretName: testsecret-tls
```

the difference is we can specify how to issue a certificate automatically at the same place. 

```
tlsClient:
    enabled: false
    #  secretName is the name of the secret that stores user-defined tidb server certificate, key and ca...
    # if not specified but tls client is enabled, certificated signed by k8s is created automatically
    # secretName: "<tidb-tls-secret-name>"
    autoGenerated:
       extraSANIPList: ...
       extraSANDomainList: ...
```

Reply:
ingress `hosts` the listen address of the http server.  A little different from ours. 

in addition, if user-defined certificate is used, they can't set `extraSAN...` here, they should set other SANs when creating certificate.

Reply:
I don't mean we need to add the `hosts` field. I just copied it from the ingress page I linked. I mean we can move `secretName` field to the up level, like `secretName` in ingress tls config. The below example is what I suggested.

Reply:
make sense

Reply:
`secretName` can have high precedence over `autoGenerated`, like `userGenerated` over `autoGenerated`. but because if a pre-created certificate is provided, I guess no extra configurations can be add in `userGenerated`, so we can replace `userGenerated` with `secretName`.

Reply:
user may use their custom Cluster TLS which used between pd, tikv and tidb peers. so there may be extra configrations?

Reply:
may other `userDefined` attributes will be added in the future that we don't know yet

Reply:
this `tlsClient` field is under `tidb` field and used to configure tidb only. if we configure tls secret for other components, we can add `secretName` in the configuration of other components.

Reply:
I mean the TLS between TiDB components: https://pingcap.com/docs/stable/how-to/secure/enable-tls-between-components/

But i am fine with only `secretName`.